### ISSUE TITLE
Training speed toooo way slow

### ISSUE THREAD

**Issue by dedoogong** [liuyuan-pal/Gen6D#16]
Hi! Thanks for your work and sharing!

I installed all the requirements and tried training "view selector" on RTX 2080Ti and V100. 
As I changed batch size to 10 and 64 for each GPU, I changed the total iteration to 60000 and 10000 separately. 
and Both case for each GPU shows way slow iteration speed as below.
I used my custom synthetic data(just 1 object and I made it as almost same as  LINEMOD format). 

1. RTX 2080Ti(batch size = 10):  14/60000 [06:52<489:44:52, 29.39s/it, loss=0.879, lr=0.0001]
2. V100 (batch size=64) : 86/10000 [1:25:35<163:57:49, 59.54s/it, loss=0.572, lr=0.0001]

Which version of CUDA/cudnn/Torch/Torchvision/Pytorch3d  do you use??
Thank you!

**Comment by liuyuan-pal** [comment]
Hi! Thanks for your interest!

The model is tested on `torch==1.10.1+cu102`, `torchversion==0.8.2`, `pytorch3d==0.3.0`. However, I think the problem is not caused by the version difference because no special models or operations are used so it should be compatible with any newer dependency versions.

The training dataset will only load 1 object per step and the `batch_size` means the number of query images on this object for training. Meanwhile, the `train_loader_batch_size` should always be 1 because current codes only support loading one object per step.

I'm not sure about the reason for your problem. Maybe you can check whether your disk IO is OK or not.

**Comment by dedoogong** [comment]
Even after i resized all imagese to 960x540, it stiill too slow. maybe there are other problems...

**Comment by liuyuan-pal** [comment]
Hi, the generation of training data requires reading many images (64 reference images and `batch_size` query images). In this case, using a large image will severely reduce the training speed. By default, all images are 128*128 for the synthetic dataset. I recommend you to crop these images before training and resize them to 128*128. Thanks!
